src/runtime_src/core/tools/xbtracer/src/lib/logger.cpp

```diff
@@ -0,0 +1,334 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#include "logger.h"
+#include "version.h"
+
+#include <filesystem>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <mutex>
+#include <array>
+
+#ifdef _WIN32
+  #include <windows.h>
+#else
+  #include <pthread.h>
+  #include <unistd.h>
+#endif /* #ifdef _WIN32 */
+
+
+namespace xrt::tools::xbtracer {
+
+constexpr unsigned int giga = 1e9;
+constexpr unsigned int hundred = 100;
+constexpr unsigned int str_sz_s = 32;
+constexpr unsigned int str_sz_m = 128;
+constexpr unsigned int str_sz_l = 256;
+constexpr unsigned int fw_9 = 9;
+#ifdef _WIN32
+constexpr std::string_view path_separator = "\\";
+#else
+constexpr std::string_view path_separator = "/";
+#endif /* #ifdef _WIN32 */
```

Comment:
There is too much ifdef this and that.   It is much better to have `detail/windows/` and `detail/linux/` directory to define the differences.   This source file (logger.cpp) should just include `#include detail/logger.h` which in turn ifdef windows and linux to include `detail/windows/logger.h` and `detail/linux/logger.h`  .  Hide the platform details in appropriate header files that are included appropriately.  See `core/common/detail` for examples.

Reply:
Done, not as cleanly as in the reference. Please review and let me know if you if this would be sufficient.